packages/components/src/internal/components/samples/SampleAliquotsGridPanel.tsx

```diff
@@ -0,0 +1,157 @@
+import React, { PureComponent } from 'react';
+
+import { PermissionTypes } from '@labkey/api';
+
+import {
+    EntityDeleteModal,
+    getSampleAliquotsQueryConfig,
+    GridPanel,
+    LoadingSpinner,
+    ManageDropdownButton,
+    QueryModel,
+    RequiresPermission,
+    SampleTypeDataType,
+    SchemaQuery,
+    SelectionMenuItem,
+    User,
+} from '../../../index';
+
+// These need to be direct imports from files to avoid circular dependencies in index.ts
+import { InjectedQueryModels, withQueryModels } from '../../../public/QueryModel/withQueryModels';
+
+import { getOmittedSampleTypeColumns } from './utils';
+
+interface Props {
+    sampleLsid: string;
+    schemaQuery: SchemaQuery;
+    user: User;
+    onSampleChangeInvalidate: (schemaQuery: SchemaQuery) => void;
+    rootLsid?: string; // if sample is an aliquot, use the aliquot's root to find subaliquots
+    storageButton?: any;
```

Comment:
`storageButton` does not accept `React.Component` typing, and errors out with the way I make a `const Node = this.props.storageButton;`. Is this the right approach?

Reply:
See https://github.com/LabKey/labkey-ui-components/pull/640#discussion_r715718431 for my recommendation as well.

Reply:
Addressed in Nick's comment!